### PR TITLE
Fixing more facets functionality and adding tests

### PR DESCRIPTION
**Pull request by devonbush**
Also refactors StudySearchProvider to have routable and non-routable components for easier testing without having to mock scpAPI. 

image

### CODE REVIEW

File: app/javascript/components/FiltersBox.js [hunk 1]
```diff
@@ -78,6 +78,7 @@ export default function FiltersBox(props) {
     if (!value) {
       _remove(newSelection, (id) => { return id === filterId; })
     }
+    debugger
```
Comment (knapii-developments) [inline app/javascript/components/FiltersBox.js]: Does this need to stay here?

File: test/js/search/more-facets.test.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import * as ReactAll from 'react';
+import { mount } from 'enzyme';
+import * as Reach from '@reach/router'
+
+const fetch = require('node-fetch');
+
+import MoreFacetsButton from 'components/MoreFacetsButton';
+import StudySearchProvider from 'components/search/StudySearchProvider';
+import * as ScpAPI from 'lib/scp-api'
```
Comment (eweitz) [inline test/js/search/more-facets.test.js]: Imports like these, that rely on Rails `webpacker.yml`, prevent developers from using out-of-the-box IDE features like `Look up definition`, `Rename symbol`, and displaying docstrings on mouse-hover.

Relative imports enable such developer productivity features by default.  Specifically:
```suggestion
import MoreFacetsButton from '../../app/javascript/components/MoreFacetsButton';
import StudySearchProvider from '../../app/javascript/components/search/StudySearchProvider';
import * as ScpAPI from '../../app/javascript/lib/scp-api'
```

File: test/js/search/more-facets.test.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+const testFacets = [{
```
Comment (eweitz) [inline test/js/search/more-facets.test.js]: Capturing our discussion on test strategy w.r.t. this approach of having mock data directly within the test and using [`const routerNav = jest.spyOn(Reach, 'navigate')`](https://github.com/broadinstitute/single_cell_portal_core/pull/434/files#diff-127e766ff6b693241ea844277b907d1dR43):

Upside:
* Tests are more self-contained in terms of readability

Downside:
* Will have false negatives upon breaking changes in other parts of our JS code, e.g. `scp-api.js`

In other words, this is towards to the unit test end of the spectrum.  

Tests in e.g. [`download.test.js`](https://github.com/broadinstitute/single_cell_portal_core/blob/b72b6048ef95adb7ae3622043b2070e48ab194c0/test/js/download.test.js) are closer to integration tests in that they cover other parts of the JS code (e.g. code in `scp-api.js`).  They're more complex to maintain, but slightly less susceptible to false negatives.

A diversity of approaches is worthwhile!

File: app/javascript/components/FiltersBox.js (hunk 1 above)
Comment (eweitz) [inline app/javascript/components/FiltersBox.js]:
```suggestion
    debugger
```

Comment (devonbush) [reply app/javascript/components/FiltersBox.js]: heh, good catch!

File: test/js/search/more-facets.test.js (hunk 2 above)
Comment (devonbush) [reply test/js/search/more-facets.test.js]: per our chat, I'll keep these for now, and investigate ways to enable IDE discovery of them

**Pull request merged**